sympy/solvers/solveset.py

```diff
@@ -223,6 +223,8 @@ def _invert_complex(f, g_ys, symbol):
         g, h = f.as_independent(symbol)
 
         if g is not S.One:
+            if g in set([S.NegativeInfinity, S.ComplexInfinity, S.Infinity]):
+                return (h, set())
```

Comment:
A SymPy `EmptySet` object should be returned here instead of the Python built-in `set`.